dtensor-keras-bert.py

```diff
@@ -263,16 +277,28 @@ def main():
 
   print("tensorflow version", tf.__version__)
 
-  # Initializes multi-client dtensor.
-  configure_virtual_cpus(8 // dtensor.num_clients())
-  dtensor.initialize_multi_client()
-
-  dprint("device type", args.device_type, "num local devices",
-         dtensor.num_local_devices(args.device_type))
+  if args.device_type != 'TPU':
```

Comment:
This branching is a bit troubling.  We probably want to unify them in the future --  not sure how to get there yet.

Quick thought:
```
initailzie_multi_client(device_type='GPU'), shall configure virtual_cpus for GPU and initialize_multi_client(device_type='TPU') shall do so for TPUs and also call initailize_tpu_system.
initailzie_multi_client(device_type='CPU'), not sure what it shall do yet -- perhaps do need num_global_devices or something like that as an input in this case.
```

Reply:
I agree that this would nicely simplify multi-client initialization. What is left to figure out is how we want users to deal with initialization when not in a multi-client setting. It doesn't make sense to have users still call `initialize_multi_client` in that case, but for TPU they would still need to insert a call to `initialize_tpu_system` somewhere. I don't know whether GPU single-client would need any setup or not.

The other solution is to have a `dtensor.initialize` that users always have to call and it is able to do all the related setup for the provided `device_type` and other env vars that are set.

Reply:
Yes. I am leaning towards an omniscient `dtensor.initialize` too. Probably eyeing towards merging into tf.context at some point.